hooks/useWeatherStations.ts

```diff
@@ -0,0 +1,129 @@
+import {useContext} from 'react';
+
+import {useQuery, useQueryClient} from 'react-query';
+
+import {ClientContext, ClientProps} from 'clientContext';
+import {ApiError, OpenAPI, StationMetadata, StationMetadataService} from 'types/generated/snowbound';
+import {AvalancheCenterID, Feature} from 'types/nationalAvalancheCenter';
+import AvalancheCenterMetadata from 'hooks/useAvalancheCenterMetadata';
+import MapLayer from 'hooks/useMapLayer';
+import {boundsForRegions, featureBounds, pointInFeature} from 'components/helpers/geographicCoordinates';
+
+type Source = 'nwac' | 'snotel' | 'mesowest';
+
+interface Props {
+  center: AvalancheCenterID;
+  sources: Source[];
+}
+
+interface ZoneResult {
+  zoneId: number;
+  name: string;
+  stationGroups: Record<string, StationMetadata[]>;
+}
+
+const stationGroupMapping = {
+  // Snoqualmie Pass
+  'Alpental Ski Area': ['1', '2', '3'],
+  'Snoqualmie Pass': ['21', '22', '23'],
+
+  // Stevens Pass
+  'Stevens Pass Ski Area - Tye Mill Chair, Skyline Chair': ['17', '18'],
+  'Stevens Pass - WSDOT Schmidt Haus': ['13'],
+  'Grace Lakes & Old Faithful': ['14', '51'],
+  'Stevens Pass Ski Area - Brooks Chair': ['50'],
+
+  // West South
+  'Crystal Mt Ski Area': ['28', '29'],
+  'Crystal Mt. - Green Valley & Campbell Basin': ['27', '54'],
+  'Mt Baker Ski Area': ['5', '6'],
+  'Mount Rainier - Sunrise': ['30', '31'],
+  'Mount Rainier - Paradise': ['35', '36'],
+  'Mount Rainier - Camp Muir': ['34'],
+  'Chinook Pass': ['32', '33'],
+  'White Pass': ['37', '39', '49'],
+  'Mt St Helens': ['40'],
+
+  // West Central
+  'White Chuck': ['57'],
+
+  // East Central
+  'Mission Ridge Ski Area': ['24', '25', '26'],
+  'Tumwater Mt. & Leavenworth': ['19', '53'],
+  'Dirtyface Mt': ['10'],
+
+  // East North
+  'Washington Pass': ['8', '9'],
+
+  // Mt Hood
+  'Skibowl Ski Area - Government Camp': ['46', '47'],
+  'Timberline Lodge': ['44', '56'],
+  'Timberline Ski Area - Magic Mile Chair': ['45'],
+  'Mt Hood Meadows Ski Area': ['42', '43'],
+  'Mt. Hood Meadows Cascade Express': ['41'],
+};
+
+const decommissionedStations = [
+  '15', // Stevens Pass - Brooks Wind (Retired 2019)
+];
+
+export const useWeatherStations = ({center, sources}: Props) => {
```

Comment:
Could you speak to why this is per-center, not per-zone?

Reply:
I'll ultimately need it to be per-center for the weather station list view, I think. This at least gives me that flexibility.